File: ast/expression/this.go
package expression

import "github.com/panda-io/micro-panda/ast/ast"

type This struct {
	ExpressionBase
}

func (t *This) Validate(c ast.Context, expected ast.Type) {
	t.Const = false
	t.Typ = c.FindObject(ast.StructThis)
	if t.Typ == nil {
		c.Error(t.GetPosition(), "undefined 'this'")
	}
}

File: MicroPanda/AST/Expression/This.cs
namespace MicroPanda.AST.Expression;

using Type;

internal class This : Expression
{
    internal override void Validate(Context context, Type? expected)
    {
        _const = false;
        _type = context.FindObject(Program.This);
        if (_type == null)
        {
            context.Program.Error(Position, "undefined 'this'");
        }
    }
}